From: Scott Ellis <scott@jumpnowtek.com>
Date: Sun, 4 Feb 2018 04:01:51 -0500
Subject: [PATCH] Add default bootargs

---
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending
Rebase for 2019.01

Index: git/include/configs/meson64.h
===================================================================
--- git.orig/include/configs/meson64.h
+++ git/include/configs/meson64.h
@@ -59,6 +59,7 @@
 
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"bootargs=console=ttyAML0,115200 rootfstype=ext4 rootwait\0" \
 	"fdt_addr_r=0x08008000\0" \
 	"scriptaddr=0x08000000\0" \
 	"kernel_addr_r=0x08080000\0" \
