From 0a3ce0c4f70f69e73b679bacdbf70076a74bf26d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 27 Mar 2018 10:28:29 -0700
Subject: [PATCH] makefile: Match the bs and count values with hardkernel
 fusing.sh script

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4cb6cde346..3fb2b83279 100644
--- a/Makefile
+++ b/Makefile
@@ -899,7 +899,7 @@ endif
 	@cp -f $(FIP_FOLDER_SOC)/u-boot.* $(FIP_FOLDER)/
 	$(Q)dd if=$(FIP_FOLDER)/u-boot.bin of=$(FUSING_FOLDER)/u-boot.bin bs=512 skip=96
 	@rm -f $(FIP_FOLDER_SOC)/boot_new.bin
-	@dd if=$(FUSING_FOLDER)/bl1.bin.hardkernel of=$(srctree)/u-boot.bin conv=fsync
+	@dd if=$(FUSING_FOLDER)/bl1.bin.hardkernel of=$(srctree)/u-boot.bin conv=fsync,notrunc bs=1 count=442
 	@dd if=$(FUSING_FOLDER)/bl1.bin.hardkernel of=$(srctree)/u-boot.bin conv=fsync,notrunc bs=512 skip=1 seek=1
 	@dd if=$(FUSING_FOLDER)/u-boot.bin of=$(srctree)/u-boot.bin conv=fsync,notrunc bs=512 seek=97
 	@echo '$(FIP_FOLDER_SOC)/u-boot.bin build done!'
-- 
2.20.1

